### DISCUSSION TITLE
Doing a Background only fit

### DISCUSSION THREAD

**Discussion by ssaha1234** [scikit-hep/pyhf#1794] [Q&A] [answered]
Dear pyhf developers/users, 
    I have an input json file, with both signal and background values, and I would like to perform two fits on it, one is background only, the other is signal + background fit. I am doing an mle fit as shown in this gist - https://gist.github.com/ssaha1234/0f49934a0806a85a261d7e61e97e8a53 

I am trying to understand how to differentiate between bkg only vs signal+ bkg fits, given I have one input json ? I am guessing the above fit result is for signal + bkg fit. 
Thank you for the help ! 

Best, 
Shreya

**Comment by kratsg** [comment, marked as answer]
> I have an input json file, with both signal and background values, and I would like to perform two fits on it, one is background only, the other is signal + background fit. I am doing an mle fit as shown in this gist - https://gist.github.com/ssaha1234/0f49934a0806a85a261d7e61e97e8a53

Typically, there's two different kinds of fits one wants to do. One is an MLE fit (unconstrained) and the other is a [POI-constrained fit](https://scikit-hep.org/pyhf/_generated/pyhf.infer.mle.fixed_poi_fit.html#pyhf.infer.mle.fixed_poi_fit). Usually when someone wants to do an MLE fit for two different models, the way it works is to do something like

```python
# ...
model = ws.model()

# bkg-only fit
pars_bonly = pyhf.infer.mle.fixed_poi_fit(0.0, ws.data(model), model)

# sig+bkg fit
pars_sigpbkg = pyhf.infer.mle.fixed_poi_fit(1.0, ws.data(model), model)
```

and you compare the fitted parameters, e.g. to compute something like

```python
model.logpdf(pars_bonly)

# vs

model.logpdf(pars_sigpbkg)
```

Does this clarify?

**Reply by alexander-held** [reply]
It's a detail, but in a fit with signal + background I would usually expect the signal normalization to free-float, i.e. `pyhf.infer.mle.fit` instead of `fixed_poi_fit`. There are certainly places where a fit with POI=1 fixed is desired as well though.

**Reply by ssaha1234** [reply]
Hi @kratsg and @alexander-held, 
   Thank you very much for the explanations. If I understand correctly, the POI constrained fit is a background-only fit `fixed_poi_fit`, and the unconstrained fit, `pyhf.infer.mle.fit` is for signal+ background.
Best, 
Shreya

**Reply by alexander-held** [reply]
The first example with `fixed_poi_fit` is a normal background-only fit, yes. You are however free to set the parameters that you hold fixed to any value you like in principle, so it does not have to be a background-only model. The second example in the comment is a fit where the signal normalization is fixed to 1, so there is signal in the model, but it is not allowed to float. `pyhf.infer.mle.fit` is unconstrained by default. You can customize that further by holding arbitrary parameters fixed at arbitrary values via the respective keyword arguments. That also allows you to reproduce the functionality of `fixed_poi_fit` via `fit`. If you look at the source code, the former is really just a convenience function to have a shortcut to achieve this specific case of setting the POI to some value, as that is quite common: https://github.com/scikit-hep/pyhf/blob/9fd99be886349a90e927672e950cc233fad0916c/src/pyhf/infer/mle.py#L199-L202

**Reply by ssaha1234** [reply]
Thanks a lot @alexander-held  for the clarification and the code snippets, it is very clear now ! 
Best, 
Shreya
